notes/2024-03-26_architecture-meeting.md

```diff
@@ -0,0 +1,99 @@
+---
+title: "2024-03-26: Architecture meeting"
+date: "2024-03-26"
+---
+
+
+## Participants
+
+* Rushiraj
+* Trey
+* Matt J.
+* Matt F.
+
+
+## Discussion
+
+### How will services get deployed at ADC?
+
+* How will NSIDC devs test/initiate deployments of services? Will NSIDC have access to
+  ADC infrastructure for developing &amp; testing code? Will we support first-class local
+  deployments so I can e.g. run the stack on my laptop?
```

Comment:
I may have missed this because I had to step away at the beginning of the meeting. @mbjones can you comment on this?

>Will we support first-class local deployments so I can e.g. run the stack on my laptop?

I believe there's good tooling for running kubernetes clusters on a local machine for development and testing, but I don't have direct experience. Do you feel this will be a challenge?

Reply:
Generally I have found that a lot of things run fine on my mac using Racher Desktop as the kubernetes implementation. The hardest part is image compatibility -- my M1 mac requires a different processor architecture for image builds, so docker images have to be built for that platform in addition to whatever linux architectures are in use. But it works.

Reply:
Looks like Rancher Desktop support Linux too, nice! Trey and I are both desktop Linux users, and Docker Desktop, last I looked, while it provides a k8s distribution, doesn't support Linux.

We ran in to this issue with Rushiraj's M1 Mac and our QGreenland Docker image! We need to learn more about how to get ARM Docker image builds in our CI...